client/components/site-icon/index.jsx

```diff
@@ -70,6 +75,19 @@ const SiteIcon = React.createClass( {
 	}
 } );
 
-export default connect( ( state, { siteId } ) => (
-	siteId ? { site: getSite( state, siteId ) } : {}
-) )( SiteIcon );
+export default connect( ( state, { site, siteId, imgSize } ) => {
+	// Until sites state is completely within Redux, we provide compatibility
+	// in cases where site object is passed to use the icon.img property as URL
+	if ( site ) {
+		return {
+			iconUrl: get( site, 'icon.img' )
+		};
+	}
+
+	// Otherwise, assume we want to perform the lookup in Redux state
+	// exclusively using the site ID
+	return {
+		site: getSite( state, siteId ),
+		iconUrl: getSiteIconUrl( state, siteId, imgSize )
```

Comment:
> If we were to render a <QuerySites />, maybe it should be limited to the condition that the site data is not yet known (and that it's passed as a site ID, i.e. Redux)?

Why is this? Is it for data usage? If so, maybe we should be able to do a new fetch just for the site info data. This would be trivial in the new API data layer since we could trap some action such as `SITE_ICON_REQUEST` and make an API call presumably to the same site settings endpoint from where it comes but then limiting the response fields to only the site icon data.